src/airflow/dags/kcca_measurements.py

```diff
@@ -14,24 +14,18 @@
 )
 def hourly_measurements_etl():
     @task(multiple_outputs=True)
-    def extract(**kwargs):
+    def extract():
         from airqo_etl_utils.date import date_to_str_hours
         from airqo_etl_utils.kcca_utils import extract_kcca_measurements
         from airqo_etl_utils.commons import fill_nan
         from datetime import datetime, timedelta
 
-        try:
-            dag_run = kwargs.get("dag_run")
-            frequency = dag_run.conf["frequency"]
-            start_time = dag_run.conf["startTime"]
-            end_time = dag_run.conf["endTime"]
-        except KeyError:
-            frequency = "hourly"
-            start_time = date_to_str_hours(datetime.utcnow() - timedelta(hours=4))
-            end_time = date_to_str_hours(datetime.utcnow())
+        hour_of_day = datetime.utcnow() - timedelta(hours=1)
+        start_date_time = date_to_str_hours(hour_of_day)
+        end_date_time = datetime.strftime(hour_of_day, "%Y-%m-%dT%H:59:59Z")
```

Comment:
Hi @NoahNsimbe. Looks like you're rounding off to the end of the hour. Is it the intended purpose?

Reply:
yes, It is intended. In an event where the start and end time are not provided, we default to start and end time of the previous hour.

Reply:
Oh, thanks